src/span.js

```diff
@@ -18,6 +100,24 @@ export default class Span {
     // ---------------------------------------------------------------------- //
 
     /**
+     * Returns the SpanContext object associated with this Span.
+     *
+     * @return {SpanContext}
+     */
+    context() {
+        if (API_CONFORMANCE_CHECKS) {
+            if (arguments.length !== 0) {
+                throw new Error('Invalid number of arguments');
+            }
+        }
+        if (this._imp) {
+            return new Tracer(this._imp.context());
+        }
+        // XXX: what to do here?
```

Comment:
I'd say this should work similarly to `Tracer.startSpan`. In other words, it should always return a `new SpanContext` and have the Span's `_imp` decide what `spanContextImp` object to pass to the interface object. I.e something like:


```
        let spanImp = null;
        if (this._imp) {
            // < removed some code here for brevity >
            spanImp = this._imp.startSpan(fields);
        }
        return new Span(spanImp);
```
https://github.com/opentracing/opentracing-javascript/blob/master/src/tracer.js#L80